### ISSUE TITLE
About the selection of negative samples

### ISSUE THREAD

**Issue by Study-gsl** [gyfastas/HCSC#5]
Hello, author. Thank you for your wonderful work. Please ask you a question. When selecting negative samples, why is the queue a randomly generated variable? Can it be set as other data samples? If I understand it incorrectly, please point it out and look forward to your answer.

self.register_buffer("queue", torch.randn(dim, queue_length))
self.queue = nn.functional.normalize(self.queue, dim=0)

**Comment by gyfastas** [comment]
Hello! I think it could be set as other data samples. However, the initialization has negligible impact on the performance because the queue would be filled by the training data quickly in a few iterations especially when the batch size is large.

----- 原始邮件 -----
发件人: "Study-gsl" ***@***.***>
收件人: "hirl-team/HCSC" ***@***.***>
抄送: "Subscribed" ***@***.***>
发送时间: 星期二, 2022年 8 月 23日 下午 4:52:51
主题: [hirl-team/HCSC] About the selection of negative samples (Issue #5)

Hello, author. Thank you for your wonderful work. Please ask you a question. When selecting negative samples, why is the queue a randomly generated variable? Can it be set as other data samples? If I understand it incorrectly, please point it out and look forward to your answer.

self.register_buffer("queue", torch.randn(dim, queue_length))
self.queue = nn.functional.normalize(self.queue, dim=0)

-- 
Reply to this email directly or view it on GitHub:
https://github.com/hirl-team/HCSC/issues/5
